modules/solid_mechanics/src/materials/IsotropicTempDepHardening.C

```diff
@@ -0,0 +1,155 @@
+
+  return params;
+}
+
+IsotropicTempDepHardening::IsotropicTempDepHardening(const InputParameters & parameters) :
+    IsotropicPlasticity(parameters),
+    _hardening_functions_names(getParam<std::vector<FunctionName> >("hardening_functions")),
+    _hf_temperatures(getParam<std::vector<Real> >("temperatures"))
+{
+  const unsigned len = _hardening_functions_names.size();
+  if (len < 2)
+    mooseError("At least two stress-strain curves must be provided in hardening_functions");
+  _hardening_functions.resize(len);
+
+  const unsigned len_temps = _hf_temperatures.size();
+  if (len != len_temps)
+    mooseError("The vector of hardening function temperatures must have the same length as the vector of temperature dependent hardening functions.");
+
+  //Check that the temperatures are strictly increasing
+  for (unsigned int i = 1; i < len_temps; ++i)
+  {
+    if (_hf_temperatures[i] <= _hf_temperatures[i-1])
+      mooseError("The temperature dependent hardening functions and corresponding temperatures should be listed in order of increasing temperature.");
+  }
+
+  std::vector<Real> yield_stress_vec;
+  for (unsigned int i = 0; i < len; ++i)
+  {
+    PiecewiseLinear * const f = dynamic_cast<PiecewiseLinear*>(&getFunctionByName(_hardening_functions_names[i]));
+    if (!f)
+      mooseError("Function " << _hardening_functions_names[i] << " not found in " << name());
+
+    _hardening_functions[i] = f;
+
+    Point p;
+    yield_stress_vec.push_back(f->value(0.0, p));
+  }
+
+  _interp_yield_stress = new LinearInterpolation(_hf_temperatures, yield_stress_vec);
+
+  _scalar_plastic_strain = &declareProperty<Real>("scalar_plastic_strain");
+  _scalar_plastic_strain_old = &declarePropertyOld<Real>("scalar_plastic_strain");
+}
+
+void
+IsotropicTempDepHardening::computeStressInitialize(unsigned qp, Real effectiveTrialStress, const SymmElasticityTensor & elasticityTensor)
+{
+  const SymmIsotropicElasticityTensor * eT = dynamic_cast<const SymmIsotropicElasticityTensor*>(&elasticityTensor);
+  if (!eT)
+    mooseError("IsotropicPlasticity requires a SymmIsotropicElasticityTensor");
+
+  _shear_modulus = eT->shearModulus();
+  initializeHardeningFunctions(qp);
+  computeYieldStress(qp);
+  _yield_condition = effectiveTrialStress - _hardening_variable_old[qp] - _yield_stress;
+  _hardening_variable[qp] = _hardening_variable_old[qp];
+  _plastic_strain[qp] = _plastic_strain_old[qp];
+}
+
+void
+IsotropicTempDepHardening::initializeHardeningFunctions(unsigned qp)
+{
+  Real temp = _temperature[qp];
+  if (temp > _hf_temperatures[0] && temp < _hf_temperatures.back())
+  {
+    for (unsigned int i = 0; i < _hf_temperatures.size() - 1; ++i)
+    {
+      if (temp >= _hf_temperatures[i] && temp < _hf_temperatures[i+1])
+      {
+        _hf_index = i;
+        Real temp_lo = _hf_temperatures[i];
+        Real temp_hi = _hf_temperatures[i+1];
+        _hf_fraction = (temp - temp_lo) / (temp_hi - temp_lo);
```

Comment:
I just realized one more thing:  You need to store per-qp values for `_hf_fraction`, `_hf_index`, and `_yield_stress` because the temperature varies per quadrature point.